docs/reference/sparql-reference.md

```diff
@@ -0,0 +1,280 @@
+# Reference templates for SPARQL queries 
+
+This document contains template SPARQL queries that can be adapted. 
+Comments are added in-code with `#` above each step to explain them so that queries can be spliced together
+
+## Checks/Report generation
+
+### All terms native to ontology 
+note: we assume that all native terms here have the same namespace - that of the ontology
+
+```SPARQL
+# select unique instances of the variable
+SELECT DISTINCT ?term
+WHERE {
+  # selecting where the variable term is either used as a subject or object 
+  { ?s1 ?p1 ?term . }
+  UNION
+  { ?term ?p2 ?o2 . }
+  # filtering out only terms that have the MONDO namespace (assumed to be native terms)
+  FILTER(isIRI(?term) && (STRSTARTS(str(?term), "http://purl.obolibrary.org/obo/MONDO_")))
+}
+```
+
+### Definition lacks xref 
+adaptable for lacking particular annotation
+
+```SPARQL
+# adding prefixes used
+prefix oio: <http://www.geneontology.org/formats/oboInOwl#>
+prefix def: <http://purl.obolibrary.org/obo/IAO_0000115>
+prefix owl: <http://www.w3.org/2002/07/owl#>
+
+SELECT ?entity ?property ?value WHERE 
+{
+  # selects only definition
+  ?entity def: ?value .
+  
+  # selecting annotation on definition
+  ?def_anno a owl:Axiom ;
+  owl:annotatedSource ?entity ;
+  owl:annotatedProperty def: ;
+  owl:annotatedTarget ?value .
+  
+  # filters out definitions which do not have a dbxref annotiton
+  FILTER NOT EXISTS {
+    ?def_anno oio:hasDbXref ?x .
+  }
+  
+  # removes triples where entity is blank
+  FILTER (!isBlank(?entity))
+  # selects entities that are native to ontology (in this case MONDO)
+  FILTER (isIRI(?entity) && STRSTARTS(str(?entity), "http://purl.obolibrary.org/obo/MONDO_"))
+  # assigning def: as the property variable
+  BIND(def: as ?property)
+}
+# arrange report by entity variable
+ORDER BY ?entity
+```
+
+### Checks wether definitions contain underscore characters 
+adaptable for checking if there is particular character in annotation
+
+```SPARQL
+# adding prefixes used
+prefix owl: <http://www.w3.org/2002/07/owl#>
+prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+prefix IAO: <http://purl.obolibrary.org/obo/IAO_>
+
+# selecting only unique instances of the three variables 
+SELECT DISTINCT ?entity ?property ?value WHERE 
+{
+  # the variable property has to be IAO:0000115 
+  VALUES ?property {
+    IAO:0000115
```

Comment:
here you use IAO:0000115, above in the query you use `def:` for the same thing. Maybe lets universally settle on `definition:`?